### PR TITLE
pi: add BinID, CacheDir, BinDir and BinPath

### PR DESCRIPTION
**Pull request by mewmew**
These functions unify handling of binary ID computation, Pippi cache directory, binary directory and binary path resolution across all Go services of Pippi.

### CODE REVIEW

File: pkg/pi/pi.go [hunk 1]
```diff
@@ -3,24 +3,77 @@ package pi
...
+func CacheDir() (string, error) {
+	cacheDir, err := os.UserCacheDir()
+	if err != nil {
+		return "", errors.WithStack(err)
+	}
+	pippiCacheDir := filepath.Join(cacheDir, "pippi")
+	return pippiCacheDir, nil
+}
+
+// BinDir returns the project directory of the given binary ID.
```
Comment (mewmew) [inline pkg/pi/pi.go:54]: @karlek Regarding terminology, should we refer to the `.cache/pippi/{BinID}/` directory as a *project directory* of a given binary ID? Or do we want to reserve *project* for later concepts and use *binary directory* instead?

Comment (karlek) [reply pkg/pi/pi.go:54]: Feels like a project is something greater. As in the example of Diablo 1, there could be multiple binaries used in a project.

File: cmd/pippi/main.go [hunk 2]
```diff
@@ -20,19 +18,12 @@ func binary(binId string) []byte {
...
 	}
-	const (
-		ext = ".bin"
-	)
-
-	cacheDir, err := os.UserCacheDir()
+	binPath, err := pi.BinPath(binId)
 	if err != nil {
-		log.Println(err)
+		log.Println()
```
Comment (karlek) [inline cmd/pippi/main.go]: Why remove the error logging?

File: cmd/pippi/main.go [hunk 3]
```diff
@@ -20,7 +20,7 @@ func binary(binId string) []byte {
 	}
 	binPath, err := pi.BinPath(binId)
 	if err != nil {
-		log.Println()
+		log.Println(err)
```
Comment (karlek) [inline cmd/pippi/main.go:23]: ah.

**Review by karlek — APPROVED** [review]
lgtm :camel: nice that we abstract away binid and not implement the wheel wherever we go

File: pkg/pi/pi.go (hunk 1 above)
Comment (mewmew) [reply pkg/pi/pi.go:54]: Yeah, I would agree. Lets reserve the term project for those uses, and revert to using "binary directory" here. I'll send a fix to *master* :)

File: cmd/pippi/main.go (hunk 2 above)
Comment (mewmew) [reply cmd/pippi/main.go]: Fat fingers :p

**Pull request merged**